openfl/component/collaborator/collaborator.py

```diff
@@ -68,18 +81,18 @@ class Collaborator:
     """
 
     def __init__(self,
-                 collaborator_name,
-                 aggregator_uuid,
-                 federation_uuid,
-                 client,
-                 task_runner,
-                 task_config,
-                 opt_treatment='RESET',
-                 device_assignment_policy='CPU_ONLY',
-                 delta_updates=False,
-                 compression_pipeline=None,
-                 db_store_rounds=1,
-                 **kwargs):
+                 collaborator_name: str,
+                 aggregator_uuid: str,
+                 federation_uuid: str,
+                 client: AggregatorGRPCClient,
+                 task_runner: CoreTaskRunner,
+                 task_config: Any,
```

Comment:
```suggestion
                 task_config: Dict[str, Dict],
```
Typing gets really hard when we are dealing with nested dictionaries. Should we create classes for such cases?